svc.go

```diff
@@ -228,3 +242,16 @@ func (as *autoNATService) background(ctx context.Context) {
 		}
 	}
 }
+
+func (as *autoNATService) extractPort(m ma.Multiaddr) (name string, port string, err error) {
+	for _, p := range m.Protocols() {
+		if p.Code == ma.P_TCP || p.Code == ma.P_UDP {
+			v, err := m.ValueForProtocol(p.Code)
```

Comment:
rather than iterating through all protocols in `m`, can you instead directly see if there is a value for the 2 protocols of instance?

Reply:
The multiformats encoding format is (<addr-protocol-code><addr-value>)+ so  ```/tcp/34372/ip4/192.168.0.100``` also a correctly formatted address, but search over bytes might be a better way.